hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -1398,6 +1398,11 @@
           // ChipWhisperer IO
           { name: 'IO_CLKOUT',        type: 'BidirStd', bank: 'VCC', connection: 'manual', desc: 'Manual clock output for SCA setup'}
           { name: 'IO_TRIGGER',       type: 'BidirStd', bank: 'VCC', connection: 'manual', desc: 'Manual trigger output for SCA setup'}
+          // SPI device debug pads
```

Comment:
are these pins supposed to connect to the spi_device module in place of the existing pins?

Reply:
Yes, that's my goal. I tried to flip what I did in #10874

Reply:
but then don't we need to make connections these new pads and the spi_device module?

Reply:
Hm this looks very similar to what needs to be done for HYP.

@tjaychen and I had a chat about this after the recent meeting where we discussed the HYP breakout. 
We think it would be best not to add more custom muxing mechanisms into the chip-level template, but rather create a second bitstream with the same RTL but a different XDC mapping. 

In the past, we had plenty such custom muxes in the FPGA chip-level hierarchy, and this was very cumbersome to maintain and prone to errors (we ran into several ASIC simulation / FPGA synthesis mismatches).
This was the main reason we added the new pinmux and chip-level templating mechanism in topgen that wires up all these pad connections automatically.

Can you maybe check whether the XDC only approach would work for you?
Also, is there any chance we can align the HYP and the dediprog use case so that they can use the same breakout / XDC?

Reply:
@msfschaffner sure. Just to be clear: The goal of this PR is to produce a bitstream that somehow lets us drive the SPI pins from outside to unblock development. I think this PR does that now after my last update but there is no intention to merge this ATM, hence the word "staging" in the title. Once we complete the implementation, we would like to keep testing this so we need a more long-term approach.

I was actually thinking about the switch/muxing approach the other day and I was on the fence:
- On one hand, it makes manual/local work somewhat easier since you can try a few things with the same bitstream.
- On the other hand, those switch configurations might end up being problematic for automated testing, i.e. someone would have to physically check/correct the switch configuration.

> Also, is there any chance we can align the HYP and the dediprog use case so that they can use the same breakout / XDC?

I think this needs more input from the HW side but for dediprog all we need is SPI lines + strap pins. HYP needs the same for bootstrap, right? Does that mean the alignment that you mentioned would be easy/trivial?

Reply:
Ah ok understood - sorry I meant this was intended to be merged.

Yes, ideally HYP will be a superset of what dediprog needs, which would make the alignment trivial since you can reuse the same bitstream / XDC.